docker/Dockerfile

```diff
@@ -0,0 +1,234 @@
+
+# Create an entrypoint that activates Emscripten SDK and helps running this image as non-root user
+COPY entrypoint ${EMSDK}/
+
+# Internal test suite of tools that this image provides
+COPY test_dockerimage.sh ${EMSDK}/
+
+
+# Create symbolic links for critical Emscripten Tools
+# This is important for letting people using Emscripten in Dockerfiles without activation
+# As each Emscripten release is placed to a different folder (i.e. /emsdk_portable/emscripten/tag-1.38.31)
+RUN echo "## Create symbolic links" \
+    &&  . ${EMSDK}/emsdk_set_env.sh \
+    \
+    &&  mkdir -p ${EMSDK}/llvm ${EMSDK}/emscripten ${EMSDK}/binaryen \
+    \
+    &&  ln -s $(dirname $(which node))/..       ${EMSDK}/node/current \
+    &&  ln -s $(dirname $(which clang))/..      ${EMSDK}/llvm/clang \
+    &&  ln -s $(dirname $(which emcc))          ${EMSDK}/emscripten/sdk \
+    \
+    &&  ln -s $(dirname $(which asm2wasm))      ${EMSDK}/binaryen/bin \
+    \
+    &&  echo "## Done"
+
+# Clean up emscripten installation and strip some symbols
+RUN echo "## Aggresive optimization: Remove debug symbols" \
+&&  apt-get -qq -y update && apt-get -qq install -y --no-install-recommends \
+        binutils \
+    && . ${EMSDK}/emsdk_set_env.sh \
+    # Remove debugging symbols from embedded node (extra 7MB)
+    && strip -s `which node` \
+    # Tests consume ~80MB disc space
+    && rm -fr ${EMSDK}/llvm/clang/emscripten/tests \
+    # strip out symbols from clang (~extra 50MB disc space)
+    && find ${EMSDK}/llvm/clang/bin -type f -exec strip -s {} + || true \
+    && find ${EMSDK}/llvm/clang/fastcomp/bin -type f -exec strip -s {} + || true \
+&&  echo "## Done"
+
+# Populate Emscripten SDK cache with libc++, to improve further compilation times.
+RUN echo "## Pre-populate cache" \
+    &&  . ${EMSDK}/emsdk_set_env.sh \
+    \
+    &&  embuilder.py build SYSTEM \
+    \
+    &&  mkdir -p /tmp/emscripten_test \
+    &&  cd /tmp/emscripten_test \
+    \
+        &&  printf '#include <iostream>\nint main(){std::cout << "HELLO FROM DOCKER C++"<<std::endl;return 0;}' > test.cpp \
+        &&  em++ --std=c++11 test.cpp -o test.js -s WASM=0 &&  node test.js \
+        &&  em++ --std=c++11 -g3 test.cpp -o test.js -s WASM=0 &&  node test.js \
+        &&  em++ --std=c++11 test.cpp -o test.js -s WASM=1 &&  node test.js \
+    \
+    &&  cd / \
+    &&  rm -fr /tmp/emscripten_test \
+    \
+    # some files were created, and we need to make sure that those can be accessed by non-root people
+    &&  chmod -R 777 ${EM_DATA} \
+    \
+    # cleanup
+    &&  find ${EMSDK} -name "*.pyc" -exec rm {} \; \
+    \
+    &&  echo "## Done"
+
+# ------------------------------------------------------------------------------
+# -------------------------------- STAGE DEPLOY --------------------------------
+# ------------------------------------------------------------------------------
+
+FROM debian:buster-slim AS stage_deploy
+
+COPY --from=stage_build /emsdk_portable /emsdk_portable
```

Comment:
How about we drop the `_portable` and just call this `/emsdk`.. I'm not sure where the portable suffix comes from.

Reply:
Good point. 
And it came from old time where emsd was distributed as a zip called `emsdk_portable` or `emsdk-portable` (I don't remember right now) and it stuck like that. 
Lots of code in this Dockerimage was transformed over 3y so, lots of legacy traces here.

Will make `/emsdk` :+1: